clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -1062,6 +1065,24 @@ private void ensureExplicitAcknowledgement() {
         }
     }
 
+    /**
+     * Initializes the acknowledgement mode based on the configuration.
+     */
+    private static AcknowledgementMode initializeAcknowledgementMode(ConsumerConfig config) {
+        if (config == null) {
+            return AcknowledgementMode.UNKNOWN;
+        }
+        String acknowledgementModeStr = config.getString(ConsumerConfig.INTERNAL_SHARE_ACKNOWLEDGEMENT_MODE_CONFIG);
+        if ((acknowledgementModeStr == null) || acknowledgementModeStr.isEmpty()) {
+            return AcknowledgementMode.UNKNOWN;
+        } else if (acknowledgementModeStr.equalsIgnoreCase("implicit")) {
+            return AcknowledgementMode.IMPLICIT;
+        } else if (acknowledgementModeStr.equalsIgnoreCase("explicit")) {
+            return AcknowledgementMode.EXPLICIT;
+        }
+        return AcknowledgementMode.UNKNOWN;
```

Comment:
Is it worth logging a warning if an unsupported value is given?

Reply:
In theory, this should not happen because of the config validation. But, I'll add it in this PR, and review this whole area when the finalised config is added. Because the config is allowed for share consumers and not allowed for regular consumers, it's a bit tricky to make sure that it validates correctly for both types of consumer.